src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php

```diff
@@ -347,8 +353,8 @@ private function publishOnExchange(\AMQPExchange $exchange, string $body, string
 
     private function setupDelay(int $delay, ?string $routingKey)
     {
-        if ($this->shouldSetup()) {
-            $this->setup(); // setup delay exchange and normal exchange for delay queue to DLX messages to
+        if ($this->autoSetupDelay) {
+            $this->setupDelayExchange(); // setup delay exchange and normal exchange for delay queue to DLX messages to
```

Comment:
ping @Nyholm is it OK? here, I setup ONLY the delay exchange, while previously you setup delayed Exchange +  normal Exchange+Queues, but it looks like this part of the code does not use the standard queues.

Reply:
hi,
I think that setup delayed exchange still not delayed really.
it maybe need to call this method `setupDelayExchange` only when getting delay exchange in this condition https://github.com/symfony/symfony/blob/f4054c1003cf97aef837cf2db5a349a87a5de5bf/src/Symfony/Component/Messenger/Bridge/Amqp/Transport/Connection.php#L371

Reply:
in a normal situation the method `declareExchange` should be called one.

Reply:
Hmm. I'm not sure to understand what you mean.

Reply:
@Nyholm I got my answer thanks to the comment on the same line :)

Reply:
> Hmm. I'm not sure to understand what you mean.

Ops, sorry, yesterday I misunderstood the code